web/api/query.go

```diff
@@ -48,6 +68,12 @@ func (serv MetricsService) Query(w http.ResponseWriter, r *http.Request) {
 	expr := params.Get("expr")
 	asText := params.Get("asText")
 
+	timestamp, err := parseTimestampOrNow(params.Get("timestamp"))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("invalid query timestamp %s", err), http.StatusBadRequest)
```

Comment:
Turns out, if one uses the `http.Error()` method, the `Content-Type` gets set to `text/plain` anyways, so this would support not returning JSON on HTTP errors.

http://golang.org/src/net/http/server.go?s=36849:36901#L1262